src/main/scala/com/dimafeng/testcontainers/TestContainer.scala

```diff
@@ -102,12 +102,48 @@ class DockerComposeContainer(composeFiles: Seq[File], exposedService: Map[String
   extends TestContainerProxy[OTCDockerComposeContainer[_]] {
 
   type OTCContainer = OTCDockerComposeContainer[T] forSome {type T <: OTCDockerComposeContainer[T]}
+
   override val container: OTCContainer = new OTCDockerComposeContainer(identifier, composeFiles.asJava)
   exposedService.foreach(Function.tupled(container.withExposedService))
 
   def getServiceHost(serviceName: String, servicePort: Int): String = container.getServiceHost(serviceName, servicePort)
 
   def getServicePort(serviceName: String, servicePort: Int): Int = container.getServicePort(serviceName, servicePort)
+
+  def withPull(pull: Boolean = true): DockerComposeContainer = {
```

Comment:
Could you please move these method to constructor parameters. The reason of it is that I'm trying to keep  containers immutable to separate creation and runtime state.